src/app/_components/statusbox.tsx

```diff
@@ -34,13 +34,17 @@ export const Statuesbox = ({ parcelslist, student_id }: Props) => {
 
   // State to track expanded parcels
   const [expandedProjectIds, setExpandedProjectIds] = useState<string[]>([]);
+  const [buttonClicked, setButtonClicked] = useState<Record<string, boolean>>(
+    {},
+  );
 
-  async function Updatestatus(project_id: string) {
-    updateparcel.mutate({
-      student_id: student_id,
-      project_id: project_id,
-    });
-  }
+  const handleButtonClick = (projectId: string) => {
+    setButtonClicked((prev) => ({ ...prev, [projectId]: true }));
+    const parcelId = parcelslist[projectId]?.[0]?.project_id;
+    // if (parcelId) {
+    //   Updatestatus(parcelId);
+    // }
+  };
```

Comment:
### Review
The change only affects the client side and does not update the data in the database. (I think you might have deleted the mutation function and didn’t use it.) Also, **don't forget to complete the mutation function** before the button color changes.

### Write code that does the following:

1. When a button is clicked,
2. Use a mutation function to update data in the database,
3. Change the button's color